File: data_samples/use/cwe_samples/390/390_Good.java

                  public String readFile(String filename) throws FileNotFoundException, IOException, Exception {
                        String retString = null;try {
                              // initialize File and FileReader objectsFile file = new File(filename);FileReader fr = new FileReader(file);
                                 // initialize character bufferlong fLen = file.length();char [] cBuf = new char[(int) fLen];
                                 // read data from fileint iRead = fr.read(cBuf, 0, (int) fLen);
                                 // close filefr.close();
                                 retString = new String(cBuf);
                              
                           } catch (FileNotFoundException ex) {System.err.println ("Error: FileNotFoundException opening the input file: " + filename );System.err.println ("" + ex.getMessage() );throw new FileNotFoundException(ex.getMessage());} catch (IOException ex) {System.err.println("Error: IOException reading the input file.\n" + ex.getMessage() );throw new IOException(ex);} catch (Exception ex) {System.err.println("Error: Exception reading the input file.\n" + ex.getMessage() );throw new Exception(ex);}return retString;
                     }
               
            

File: data_samples/use/cwe_samples/390/390_Good.cpp

                  char* readFile (char *filename) {
                        try {
                              // open input fileifstream infile;infile.open(filename);
                                 if (!infile.is_open()) {throw "Unable to open file " + filename;}
                                 // get length of fileinfile.seekg (0, ios::end);int length = infile.tellg();infile.seekg (0, ios::beg);
                                 // allocate memorychar *buffer = new char [length];
                                 // read data from fileinfile.read (buffer,length);
                                 if (!infile.good()) {throw "Unable to read from file " + filename;}infile.close();
                                 return buffer;
                           }catch (char *str) {printf("Error: %s \n", str);infile.close();throw str;}catch (...) {printf("Error occurred trying to read from file \n");infile.close();throw;}
                     }
               
            